model/patch_lifecycle.go

```diff
@@ -524,18 +524,54 @@ func MakeCommitQueueDescription(patches []patch.ModulePatch, projectRef *Project
 	return "Commit Queue Merge: " + strings.Join(description, " || ")
 }
 
-func MakeMergePatchFromExisting(existingPatchID string) (*patch.Patch, error) {
-	existingPatch, err := patch.FindOne(patch.ById(patch.NewId(existingPatchID)))
+type EnqueuePatch struct {
+	PatchID string
+}
+
+func (e *EnqueuePatch) String() string {
+	return fmt.Sprintf("enqueue patch '%s'", e.PatchID)
+}
+
+func (e *EnqueuePatch) Send() error {
+	existingPatch, err := patch.FindOneId(e.PatchID)
 	if err != nil {
-		return nil, errors.Wrapf(err, "problem getting existing patch '%s'", existingPatchID)
+		return errors.Wrap(err, "can't get existing patch")
 	}
 	if existingPatch == nil {
-		return nil, errors.Errorf("patch '%s' doesn't exist", existingPatchID)
+		return errors.Errorf("no patch '%s' found", e.PatchID)
+	}
+
+	// only enqueue once
+	if len(existingPatch.MergePatch) != 0 {
+		return nil
 	}
 
+	cq, err := commitqueue.FindOneId(existingPatch.Project)
+	if err != nil {
+		return errors.Wrap(err, "can't get commit queue")
+	}
+	if cq == nil {
+		return errors.Errorf("no commit queue for project '%s'", existingPatch.Project)
+	}
+
+	mergePatch, err := MakeMergePatchFromExisting(existingPatch)
+	if err != nil {
+		return errors.Wrap(err, "problem making merge patch")
+	}
+
+	_, err = cq.Enqueue(commitqueue.CommitQueueItem{Issue: mergePatch.Id.Hex()})
```

Comment:
Do we need to worry about modules anywhere, or do we already worry about them? (That could be another ticket though if we do need something)

Reply:
My plan for modules is that if they're part of the original commit queue patch they'll be part of the backport as well.